core/vulkan/vk_virtual_swapchain/cc/virtual_swapchain.cpp

```diff
@@ -353,27 +381,35 @@ void VirtualSwapchain::CopyThreadFunc() {
       }
     }
 
-    VkResult ret = functions_->vkWaitForFences(
+    VkResult res = functions_->vkWaitForFences(
         device_, 1, &image_data_[pending_image].fence_, false, UINT64_MAX);
-    (void)ret;  // TODO: Check this?
-    functions_->vkResetFences(device_, 1, &image_data_[pending_image].fence_);
+    LOG_IF_FAILED(res, "CopyThreadFunc: vkWaitForFences");
```

Comment:
In general I would prefer an idiom more like 
```
VkResult res = EXPECT_SUCCESS(functions_->vkWaitForFences(
        device_, 1, &image_data_[pending_image].fence_, false, UINT64_MAX);

#define EXPECT_SUCCESS(...) \
    ({ auto r = __VA_ARGS__; if (r != 0) { printf(#__VA_ARGS__);}  r;})
```
Or if you want it to work in MSVC as well, wrap that in a lambda

This will output the exact code that failed, (You can also stick in a line/file if you want)

Reply:
OK thanks. I have done something like this now. The warnings look like this:

```
VirtualSwapchainLayer: core/vulkan/vk_virtual_swapchain/cc/virtual_swapchain.cpp:225: functions_->vkAllocateMemory( device_, &image_memory_info, pAllocator, &image_data.image_memory_) RETURNED: -2
```